File: OOP/1_intro/Reckless+Driver+-+C++/RecklessDriver/RecklessDriver/Van.cpp
#include "Van.h"
#include <iostream>
#include "Player.h"
namespace RecklessDriver {

	Van::Van(int _damage, int _cash) :Vehicle(_damage, _cash)
	{
		SetName("Van");
	}

	void Van::OnCollision(GameObject *other)
	{
		if (other->GetTag() == "Player") {
			Player *p = (Player *)other;
			std::cout << "#### COLLISION ->[Van] Milk bottles falling" << std::endl;
			p->ApplyDamage(GetDamage(), GetCash());
		}
	}
}

File: OOP/1_intro/Reckless+Driver+-+Java/Reckless Driver/src/com/poash/rd/Van.java
package com.poash.rd;

public class Van extends Vehicle {

	public Van(int damage, int cash) {
		super(damage, cash);
		setName("Van");
	}
	@Override
	public void OnCollision(GameObject other) {
		if(other.getTag().equals("Player")){
			Player player = (Player)other; 
				System.out.println("#### COLLISION ->[Van] Milk bottles falling");
				player.ApplyDamage(getDamage(), getCash());
		}
	}

}
